client/src/OnDeviceComponent.ts

```diff
@@ -467,7 +467,15 @@ export class OnDeviceComponent {
 	}
 
 	private setupClientSocket(options: ODC.RequestOptions) {
-		return new Promise<net.Socket>((resolve, reject) => {
+		if (this.clientSocket) {
+			return this.clientSocket;
```

Comment:
This doesn't look right. A function should always return a promise, or always return a non promise. You're mixing the actual socket return value and promise return value. 

This should probably be:
```
return Promise.resolve(this.clientSocket);
```